functions/Get-DbaMemoryUsage.ps1

```diff
@@ -0,0 +1,137 @@
+        $Memcounters = '(Total Server Memory |Target Server Memory |Connection Memory |Lock Memory |SQL Cache Memory |Optimizer Memory |Granted Workspace Memory |Cursor memory usage|Maximum Workspace)'
+        $Plancounters = '(cache pages|procedure plan|ad hoc sql plan|prepared SQL Plan)'
+        $BufManpagecounters = '(Free pages|Reserved pages|Stolen pages|Total pages|Database pages|target pages|extension .* pages)'
+        }
+
+    }
+
+	PROCESS
+	{
+        foreach ($servername in $ComputerName)
+        {
+            if ($servername -match '\\')
+			{
+				$servername = $servername.Split('\\')[0]
+			}
+            Write-Verbose "Connecting to $servername"
+			if ( test-connection -ComputerName $servername -Quiet -count 1)
+            {
+                $availablecounters = (get-counter -ComputerName $servername -ListSet '*sql*:Memory Manager*').paths
+                (get-counter -ComputerName $servername -Counter $availablecounters).countersamples | 
+                    Where-Object {$_.Path -match $Memcounters} | 
+                    foreach { [PSCustomObject]@{
+				                ComputerName = $servername
+				                Instance = $_.Path.split("\")[-2]
+                                Counter = $_.Path.split("\")[-1]
+				                Pages = $null
+				                MemKB = $_.cookedvalue
+				                MemMB = $_.cookedvalue / 1024
+                                }
+                            }
+
+
+                $availablecounters = (get-counter -ComputerName $servername -ListSet '*sql*:Plan Cache*' ).paths
+                (get-counter -ComputerName $servername -Counter $availablecounters).countersamples |
+                    Where-Object {$_.Path -match $Plancounters} |
+                    foreach { [PSCustomObject]@{
+					            ComputerName = $servername
+					            Instance = $_.Path.split("\")[-2]
+                                Counter = $_.Path.split("\")[-1]
+					            Pages = $_.cookedvalue
+					            MemKB = $_.cookedvalue * 8192 / 1024
+					            MemMB = $_.cookedvalue * 8192 / 1048576
+                                }
+                            }
+
+
+                $availablecounters = (Get-Counter -ComputerName $Servername -ListSet "*Buffer Manager*").paths
+                (get-counter -ComputerName $Servername -Counter $availablecounters).countersamples |
+                    Where-Object {$_.Path -match $BufManpagecounters} |
+                    foreach { [PSCustomObject]@{
+					            ComputerName = $servername
+					            Instance = $_.Path.split("\")[-2]
+                                Counter = $_.Path.split("\")[-1]
+					            Pages = $_.cookedvalue
+					            MemKB = $_.cookedvalue * 8192 / 1024.0
+					            MemMB = $_.cookedvalue * 8192 /1048576.0
+                                }
+                            }
+            }
+			else
+			{
+				Write-Warning "Can't connect to $servername. Moving on."

```

Comment:
I don't think so: the test-connection is inside the if clause. If this returns nothing, the if {} block is skipped and the next step is the else {} block.